### PR TITLE
Add 3D support for XAI methods and fix Grad-CAM decoder issues

### LINKED ISSUE #233: xai decoder issue gradcam gradient access
Inputs and outputs of gradient access model part of the gradient-based XAI methods have to be reworked due to keras 3 now enforces keras tensors and do not allow lists anymore.

Fix:
```py
        # Gradient model construction
        layer_output = self.model.get_layer(self.layerName).output
        model_output = self.model.output
        if isinstance(model_output, list):
            outputs = [layer_output] + model_output
        else:
            outputs = [layer_output, model_output]

        gradModel = tf.keras.models.Model(inputs=self.model.inputs,
                         outputs=outputs)
```

todo: make PR

### LINKED ISSUE #234: xai decoder return image with unequal shape
Original: aucmedi/xai/decoder.py
```py
    # Return output directly if no output path is defined (and convert to NumPy)
    if out_path is None : return np.array(res_img), np.array(res_xai)

```

Fix:
```py 
    # Return output directly if no output path is defined (and convert to NumPy)
    if out_path is None : return res_img, res_xai
```

Also update docstring

### PR DESCRIPTION
**Pull request by rizoudal**
This PR introduces full 3D support for multiple XAI methods in AUCMEDI and resolves issues related to Grad-CAM and decoder output handling.

### Merged `experimental.3D_gradcam` branch
- Integrated 3D Grad-CAM implementation and volume visualizer
- Added unit tests for volume-based inputs 

### Fixed issue #233
- Corrected gradient model construction in Grad-CAM
- Applied equivalent fix to Grad-CAM++

### Added 3D support for:
- Grad-CAM++
- Guided Backpropagation
- Guided Grad-CAM
- Saliency Maps

Each method includes dedicated unit tests for both HU and RGB 3D volumes.

### Fixed issue #234
- Removed `np.array()` conversion in `xai_decoder` return values
- Adapted unit tests to handle native image objects as returned by decoder

---

## Related Issues

- Closes #233  
- Closes #234

### CODE REVIEW

File: aucmedi/data_processing/data_generator.py [hunk 1]
```diff
@@ -256,14 +256,15 @@ def __init__(self, samples, path_imagedir, labels=None, metadata=None,
             if self.workers == 0 or self.workers == 1:
                 for i in range(0, len(samples)):
                     self.preprocess_image(index=i, prepared_image=False,
-                                          run_aug=False, run_standardize=False,
+                                          run_resize=True, run_aug=False,
```
Comment (muellerdo) [inline aucmedi/data_processing/data_generator.py:259]: run_resize?

File: aucmedi/data_processing/data_generator.py
Comment (muellerdo) [inline aucmedi/data_processing/data_generator.py:1]: why adding of run_resize necessary?

File: aucmedi/xai/decoder.py [hunk 2]
```diff
@@ -141,13 +164,18 @@ def postprocess_output(sample, image, xai_map, n_classes, data_gen,
...
+                              out_path=path_xai, alpha=alpha)
+        # If no preds given (sample_maps), output heatmaps for all classes
         else:
             for c in range(0, n_classes):
-                path_xai_c = path_xai[:-4] + ".class_" + str(c) + \
-                             path_xai[-4:]
-                visualize_heatmap(image, xai_map[c], out_path=path_xai_c,
-                                  alpha=alpha)
+                if data_gen.image_format is None:
+                    ff = len(xai_file.split(".")[-1]) + 1
```
Comment (muellerdo) [inline aucmedi/xai/decoder.py:175]: what is happening here?

**Review by muellerdo — COMMENTED** [review]
Discussion of open questions.
